### PR TITLE
CA-352329: Revert original formatting of lists in xapi-cli-server

### PR DESCRIPTION
**Pull request by psafont**
Also define functions so conventions can be easily followed

### CODE REVIEW

**Review by edwintorok — APPROVED** [review]
I compared this to reverting the 3 xapi-cli-server commits on master and wherever `",` was used I use the `comma` variant used, so LGTM

**Comment by edwintorok** [comment]
Except maybe for roles where we might now show nothing and previously we only skipped the item that was failing:
```
       ; make_field ~name:"roles"
           ~get:(fun () ->
-            String.concat "; "
-              ( try
-                  List.map
+            try
+              map_and_concat
                 (fun self ->
                   try Client.Role.get_name_label rpc session_id self
                   with _ -> nid)
                 (Client.Subject.get_roles rpc session_id subject)
-                with _ -> []
-              ))
+            with _ -> "")

```

File: ocaml/xapi-cli-server/records.ml [hunk 1]
```diff
@@ -168,10 +168,16 @@ let get_uuid_from_ref_or_null r =
 
 let concat_with_semi = String.concat "; "
 
+let concat_with_comma = String.concat ", "
```
Comment (robhoes) [inline ocaml/xapi-cli-server/records.ml:171]: Let's add a comment to discourage new uses of this function, `map_and_concat_comma` and `get_uuids_from_refs_outdated` (explaining why).

Comment (robhoes) [reply ocaml/xapi-cli-server/records.ml:173]: That was quick...

Comment (psafont) [reply ocaml/xapi-cli-server/records.ml:173]: ;)

**Review by robhoes — APPROVED** [review]

**Pull request merged**